File: 555plus999.js
#!/usr/bin/env node

// 555 + 999 计算脚本

console.log("=== 555 + 999 计算脚本 ===\n");

// 方法1: 直接相加
const result = 555 + 999;
console.log("方法1 - 直接相加: 555 + 999 = " + result);

// 方法2: 分解计算
const a = 555;
const b = 999;
console.log("\n方法2 - 分解计算:");
console.log("  " + a + " + " + b);
console.log("  = " + a + " + (1000 - 1)");
console.log("  = " + a + " + 1000 - 1");
console.log("  = " + (a + 1000) + " - 1");
console.log("  = " + result);

// 方法3: 位运算演示 (仅用于展示)
console.log("\n方法3 - 验证结果:");
console.log("  555 的二进制: " + 555.toString(2));
console.log("  999 的二进制: " + 999.toString(2));
console.log("  结果的二进制: " + result.toString(2));
console.log("  结果转十进制验证: " + parseInt(result.toString(2), 2) + " (应等于 " + result + ")");

console.log("\n=== 最终结果: 555 + 999 = " + result + " ===");

File: 555plus999.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

# 555 + 999 计算脚本

print("=== 555 + 999 计算脚本 ===\n")

# 方法1: 直接相加
result = 555 + 999
print(f"方法1 - 直接相加：555 + 999 = {result}")

# 方法2: 分解计算
a = 555
b = 999
print(f"\n方法2 - 分解计算:")
print(f"  {a} + {b}")
print(f"  = {a} + (1000 - 1)")
print(f"  = {a} + 1000 - 1")
print(f"  = {a + 1000} - 1")
print(f"  = {result}")

# 方法3: 二进制演示
print(f"\n方法3 - 验证结果:")
print(f"  555 的二进制：{bin(555)}")
print(f"  999 的二进制：{bin(999)}")
print(f"  结果的二进制：{bin(result)}")
print(f"  结果转十进制验证：{int(bin(result), 2)} (应等于 {result})")

print(f"\n=== 最终结果：555 + 999 = {result} ===")
